myproject/app.py

```diff
@@ -0,0 +1,272 @@
+from flask_bcrypt import Bcrypt
+from datetime import datetime
+from flask_migrate import Migrate
+from model import  db , user, post , comment , Follow , Like
+from datetime import datetime
+from utlis import add_user , new_post , delete , save_changes , add
+from config import basesit
+
+
+
+app = Flask(__name__)
+app.config.from_object(basesit)
+db.init_app(app)
+jwt = JWTManager(app)
+bcrypt = Bcrypt(app)
+migrate = Migrate(app, db)
+date = datetime.now()
+#User Management 
+@app.route("/register", methods=['POST'])
+def register_user():
+    data = request.json
+    email = data.get('email')
+    username = data.get('username')
+    password = data.get('password')
+    if not all([email, username, password]):
+        return jsonify({'error': 'All fields need to be provided'}), 400
+    if user.query.filter_by(email=email).first() or user.query.filter_by(username=username).first():
+        return jsonify({'error': 'User with this email or username already exists'}), 409
+    hashed_password = bcrypt.generate_password_hash(password).decode('utf-8')
+    add_user(email,username , hashed_password)
+    return jsonify({'message': 'User created successfully'}), 201
+
+@app.route('/login', methods=['POST'])
+def login():
+    data = request.get_json()
+    email = data.get('email')
+    password = data.get('password')
+    User = user.query.filter_by(email=email).first()
+    if not User:
+        return jsonify({'message': 'User not found'}), 404
+    if bcrypt.check_password_hash(User.password, password):
+        access_token = create_access_token(identity={'username': User.username, 'email': User.email, 'id': User.id})
+        return jsonify({'access_token': access_token}), 200
+    else:
+        return jsonify({'message': 'Invalid credentials'}), 401
+
+@app.route('/retrieve/<int:id>', methods=['GET'])
+@jwt_required()
+def get_user(id):
+    User = user.query.get(id)
+    if User:
+        return jsonify({'username': User.username, 'email': User.email})
+    else:
+        return jsonify({'message': 'User not found'})
+
+@app.route("/update/<int:id>", methods=['PUT'])
+@jwt_required()
+def update_user(id):
+    User = user.query.get(id)
+    if not User:
+        return jsonify({'error': 'User not found'})
+    data = request.json
+    User.email = data.get('email', User.email)
+    User.username = data.get('username', User.username)
+    db.session.commit()
+    return jsonify({'message': 'User updated successfully'})
+
+@app.route("/delete/<int:id>", methods=['DELETE'])
+@jwt_required()
+def delete_user(id):
+    User = user.query.get(id)
+    if not User:
+        return jsonify({'error': 'User not found'}), 404
+    db.session.delete(User)
+    db.session.commit()
+    return jsonify({'message': 'User deleted successfully'})
+
+
+# Post Management
+@app.route("/create_blog", methods=['POST'])
+@jwt_required()
+def create_blog():
+    data = request.json
+    title = data.get('title')
+    author = get_jwt_identity()['username'] 
```

Comment:
If the username is stored in the author, what is the purpose of the user ID in the model?